iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/LineAdderImpl.java

```diff
@@ -110,6 +122,26 @@ public LineImpl add() {
         line.addTerminal(terminal1);
         line.addTerminal(terminal2);
 
+        if (copiedLine != null) {
+            Collection<Extension<Line>> extensions = copiedLine.getExtensions();
+            extensions.forEach(lineExtension ->
+                   line.addExtension((Class<? super Extension<Line>>) lineExtension.getClass(), lineExtension));
```

Comment:
Beware: here it does not copy the extension. It modifies it to point to `line` and adds it to `line. Thus, after calling `add()` on the adder, `line` and `copiedLine` will have the same Java object as extension.